src/camp/generators/data/agent.c.jinja

```diff
@@ -66,9 +66,9 @@ Produced type: {{type_summary(obj.typeobj)}}
 {
     cace_amm_formal_param_t *fparam = refda_register_add_param(obj, {{param.name | c_str}});
     {{amm_type_set('&(fparam->typeobj)', param.typeobj) | indent(4)}}
-    {%- if param.default_value %}
+{%- if param.default_ari %}
```

Comment:
**Observation**: This may be a separate issue but it seems we should check if the typeobj and default ARI are compatible. For example if we try to initialize a `BYTESTR` parameter to a floating point value:

```
                cace_amm_formal_param_t *fparam = refda_register_add_param(obj, "object");
                {   
                    ari_t name = ARI_INIT_UNDEFINED;
                    ari_set_aritype(&name, ARI_TYPE_BYTESTR);
                    amm_type_set_use_ref_move(&(fparam->typeobj), &name);
                }
                ari_set_real64(&(fparam->defval), 3.100000e+01);
```